includes/vendor/mautic.php

```diff
@@ -0,0 +1,607 @@
+<?php
+
+/**
+ * Helper class for the Mautic API.
+ *
+ * @since 1.0.5
+ */
+	class AP_MauticAPI {
+
+	/**
+	 * @since 1.5.4
+	 * @var object $api_instance
+	 * @access private
+	 */
+	private $api_instance = null;
+
+	/**
+	 * Update Mautic credentials
+	 *
+	 * @since 1.0.0
+	 * @return void
+	 */
+	public function get_mautic_data() {
+		$credentials = APMautic_helper::get_mautic_credentials();
+		// If not authorized.
+		if ( ! isset( $credentials['access_token'] ) ) {
+			if ( isset( $credentials['access_code'] ) ) {
+				$grant_type = 'authorization_code';
+				$response = self::mautic_get_access_token( $grant_type );
+
+				if ( is_wp_error( $response ) || 200 !== wp_remote_retrieve_response_code( $response ) ) {
+					echo __( 'There appears to be an error with the configuration.', 'automateplug-mautic-wp' );
+					$status   = 'error';
+				} else {
+
+					$response_body = wp_remote_retrieve_body( $response );
+
+					$access_details               = json_decode( $response_body );
+					$expiration                   = time() + $access_details->expires_in;
+					$credentials['access_token']  = $access_details->access_token;
+					$credentials['expires_in']    = $expiration;
+					$credentials['refresh_token'] = $access_details->refresh_token;
```

Comment:
All the parameters received from the API should be sanitized before saving to the database.